pkg/pb/piecestore2.proto

```diff
@@ -90,3 +90,23 @@ message RetainRequest {
 
 message RetainResponse {
 }
+
+// PieceHeader is used in piece storage to keep track of piece attributes.
+message PieceHeader {
+    // the storage format version being used for this piece. The piece filename should agree with this.
+    // The inclusion of this field is intended to aid repairability when filenames are damaged.
+    int32 format_version = 1;
+    // content hash of the piece
+    bytes hash = 2;
+    // timestamp when upload occurred, as given by the "timestamp" field in the original orders.PieceHash
+    google.protobuf.Timestamp creation_time = 3 [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
+    // signature from uplink over the original orders.PieceHash (the corresponding PieceHashSigning
+    // is reconstructable using the piece id from the piecestore, the piece size from the
+    // filesystem (minus the piece header size), and these (hash, upload_time, signature) fields).
+    bytes signature = 4;
+    // expiration timestamp, as given in the OrderLimit
+    google.protobuf.Timestamp expiration_time = 5 [(gogoproto.stdtime) = true];
```

Comment:
Since we are storing the `orders.OrderLimit` also, why store the `expiration_time` in its own field? Are we protecting against a future date when we are not storing the order limit or something?

Reply:
Good catch. I only had expiration_time first, and then when I added the order limit I didn't notice that it already had expiration_time in it. Will fix.